src/commands/types.ts

```diff
@@ -0,0 +1,78 @@
+import { CSharpFileGenerator, JavaFileGenerator, JavaScriptFileGenerator } from '@asyncapi/modelina';
+import { flags } from '@oclif/command';
+import Command from '../base';
+import { load } from '../models/SpecificationFile';
+import { parse } from '@asyncapi/parser';
+export default class GenerateTypes extends Command {
+  static description = 'Generates typed models';
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    file: flags.string({ char: 'f', description: 'path to the AsyncAPI file to generate types for' }),
+    outputDirectory: flags.string({ char: 'o', description: 'output path to write the types to', required: true}),
+    language: flags.enum({ char: 'l', description: 'language output', options: ['typescript', 'csharp', 'golang', 'java', 'javascript'], required: true}),
+    /**
+     * Go specific options
+     */
+    goPackageName: flags.string({description: 'Go specific, define the package to use for the generated types', required: false}),
+    /**
+     * Java specific options
+     */
+    javaPackageName: flags.string({description: 'Java specific, define the package to use for the generated types', required: false}),
+    /**
+     * C# specific options
+     */
+    csharpNamespace: flags.string({description: 'C# specific, define the namespace to use for the generated types', required: false}),
+  }
+
+  static args = []
+
+  async run() {
+    const { flags } = this.parse(GenerateTypes);
+    const outputDirectory = flags.outputDirectory;
+    const file = await load(flags.file) || await load();
+    const parsedInput = await parse(await file.read());
+    const language = flags.language;
+    let fileGenerator;
+    let fileOptions = {};
+    if (language === 'typescript') {
```

Comment:
I know that this is a draft PR, but maybe you should use `switch case` with separate function for every language? That small suggestion 😄

Reply:
Changed 🙂

Reply:
Where? 😄

Reply:
In my local commit-tree 😆

Reply:
Finishing the PR before pushing the changes.